src/ansys/sound/core/signal_processing/filter.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Filter a signal."""
+
+import warnings
+
+from ansys.dpf.core import Field, Operator
+import matplotlib.pyplot as plt
+import numpy as np
+
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+from ..signal_processing import SignalProcessingParent
+
+ID_OPERATOR_DESIGN = "filter_design_minimum_phase_FIR_filter_from_FRF"
+ID_OPERATOR_LOAD = "load_FRF_from_txt"
+ID_OPERATOR_FILTER = "filter_signal"
+
+
+class Filter(SignalProcessingParent):
```

Comment:
@ansaminard in the specs we specified that the parent of `Filter` should be `SpectralProcessingParent`

Now, I can understand that `SignalProcessingParent` could be meaningful as well, what's your take on this ? 

@anshlachamb same question :)

Reply:
@a-bouth What? Are you sure your version of the branch is up to date?

Reply:
@ansaminard ignore that comment, I didn't see @anssmolla's comment